Startup.cs

```diff
@@ -27,31 +21,7 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc();
 
-            services.AddAuthentication(options =>
-            {
-                options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
-                options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
-            }).AddJwtBearer(o =>
-            {
-                o.Authority = Configuration["Jwt:Authority"];
-                o.Audience = Configuration["Jwt:Audience"];
-                o.RequireHttpsMetadata = false; //TODO remove this in production
-                o.Events = new JwtBearerEvents()
-                {
-                    OnAuthenticationFailed = c =>
-                    {
-                        c.NoResult();
-
-                        c.Response.StatusCode = 500;
-                        c.Response.ContentType = "text/plain";
-                        if (Environment.IsDevelopment())
-                        {
-                            return c.Response.WriteAsync(c.Exception.ToString());
-                        }
-                        return c.Response.WriteAsync("An error occured processing your authentication.");
-                    }
-                };
-            });
+            StartupAuth.ConfigureServices(services, Configuration, Environment);
```

Comment:
I like the idea to move the complicated Auth setup to its own class. I think it would be better to add it as an extension method, this way we keep the "standard" structure, and we can call `services.AddAzureB2CAuthentication(...).
I'm not sure I like passing the whole `Environment` and the whole `Configuration` as well. We should pass only what the configuration needs.

Reply:
I'm open.  I didn't like the implementation of my approach.

I like small specific classes.  The Asp.Net core team seems too except for the startup file.  I haven't seen and samples/recommendations on how to keep it clean!

I'll update the code for the extension method + passing only what is needed.

I wish we could just use DI...

Reply:
Yeah, startup class can become a mess quickly if you aren't careful. I really like your approach with creating small static classes do set that up, and you can see that it's also the MS team approach (`services.AddMvc()` for example). I'm definitely going to steal your idea to cleanup some of my other code 😄.

It's just that I think it would look better with the extension method, since it'll be also the same as `services.AddMvc()` and other existing extension methods from the base framework. For the Configuration and Environment it's just that I think it's always better to pass the most precise object. If there are too much information to pass, you can create a small class to pass as Configuration (like the `DevPackMiddleWareOptions` class for example).

Reply:
It sure can become a mess.  Thanks!!  Ah, yes they do!  Now I see it 🎉 !!

Reply:
Yes, I find it pretty clever how they use extension methods and DI to easily make plugins add functions to .Net Core. It makes things very clean.